code/game/gamemodes/survivor.dm

```diff
@@ -0,0 +1,460 @@
+    var/xeno_ratio = 0.3 // otherwise fill to ratio of xeno to humans
+
+    var/list/key_items = list( // Beacon requires repair at lowpop
+        /obj/item/beacon/rescue,
+        /obj/item/tool/weldingtool,
+        /obj/item/cell/high,
+        /obj/item/laptop/rescue,
+    )
+
+    var/list/random_items = list() // Use to add more random items to the map
+
+/datum/game_mode/survivor/new_player_panel(mob/new_player/NP)
+
+    var/output = "<div align='center'>"
+    output += "<p><a href='byond://?src=[REF(NP)];lobby_choice=show_preferences'>Setup Character</A></p>"
+
+    if(SSticker.current_state <= GAME_STATE_PREGAME)
+        output += "<p>\[ [NP.ready? "<b>Ready</b>":"<a href='byond://?src=\ref[src];lobby_choice=ready'>Ready</a>"] | [NP.ready? "<a href='byond://?src=[REF(NP)];lobby_choice=ready'>Not Ready</a>":"<b>Not Ready</b>"] \]</p>"
+    else
+        output += "<p><a href='byond://?src=[REF(NP)];lobby_choice=late_join_survivor'>Join the survivors!</A></p>"
+        output += "<p><a href='byond://?src=[REF(NP)];lobby_choice=late_join_xeno'>Join the Hive!</A></p>"
+
+    output += "<p><a href='byond://?src=[REF(NP)];lobby_choice=observe'>Observe</A></p>"
+
+    if(!IsGuestKey(NP.key))
+        if(SSdbcore.Connect())
+            var/isadmin = FALSE
+            if(check_rights(R_ADMIN, FALSE))
+                isadmin = TRUE
+            var/datum/DBQuery/query_get_new_polls = SSdbcore.NewQuery("SELECT id FROM [format_table_name("poll_question")] WHERE [(isadmin ? "" : "adminonly = false AND")] Now() BETWEEN starttime AND endtime AND id NOT IN (SELECT pollid FROM [format_table_name("poll_vote")] WHERE ckey = \"[sanitizeSQL(NP.ckey)]\") AND id NOT IN (SELECT pollid FROM [format_table_name("poll_textreply")] WHERE ckey = \"[sanitizeSQL(NP.ckey)]\")")
+            if(query_get_new_polls.Execute())
+                var/newpoll = FALSE
+                if(query_get_new_polls.NextRow())
+                    newpoll = TRUE
+
+                if(newpoll)
+                    output += "<p><b><a href='byond://?src=[REF(NP)];showpoll=1'>Show Player Polls</A> (NEW!)</b></p>"
+                else
+                    output += "<p><a href='byond://?src=[REF(NP)];showpoll=1'>Show Player Polls</A></p>"
+            qdel(query_get_new_polls)
+            if(QDELETED(src))
+                return
+
+    output += "</div>"
+
+    var/datum/browser/popup = new(src, "playersetup", "<div align='center'>New Player Options</div>", 240, 300)
+    popup.set_window_options("can_close=0")
+    popup.set_content(output)
+    popup.open(FALSE)
+
+
+/datum/game_mode/survivor/new_player_topic(mob/M, href, href_list[])
+    switch(href_list["lobby_choice"])
+        if("late_join_survivor")
+            if (world.time > (SSticker.round_start_time + late_join_time))
+                to_chat(src, "<span class='warning'>It's too late to join this round.</span>")
+                return
+
+            if (!M.mind)
```

Comment:
Every mob should have a mind at all times, it may be better to rather stack_trace here

Reply:
Added the `stack_trace`